packages/css-selector-parser/src/ast-tools/walk.ts

```diff
@@ -51,10 +69,10 @@ export function walk(
       // visit node
       let skipAmount =
         (visit(
-          current,
+          current as SelectorNode,
```

Comment:
is this because the visitor signature is not known inside the function?

Reply:
yes, visitor signature is overloaded with a union of `WalkVisitor<SelectorNode> | WalkVisitor<ImmutableSelectorNode>` and I couldn't get this to work in any other way :/